investment-app/TRINV.Api/TRINV.Application/Queries/UserNotifications/GetAllUserNotificationsQuery.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace TRINV.Application.Queries.UserNotifications;
+
+using Domain.Entities;
+using Interfaces;
+using MediatR;
+using Shared.Business.Exceptions;
+using Shared.Business.Extension;
+using Shared.Business.Utilities;
+
+public record GetAllUserNotificationsQuery : IRequest<OperationResult<IEnumerable<UserNotification>>>;
+
+internal class GetAllUserNotificationsQueryHandler : IRequestHandler<GetAllUserNotificationsQuery, OperationResult<IEnumerable<UserNotification>>>
+{
+    readonly IRepository<UserNotification> _userNotificationRepository;
+    readonly IUserContext _userRepository;
+
+    public GetAllUserNotificationsQueryHandler(IRepository<UserNotification> userNotificationRepository, IUserContext userRepository)
+    {
+        _userNotificationRepository = userNotificationRepository;
+        _userRepository = userRepository;
+    }
+
+    public async Task<OperationResult<IEnumerable<UserNotification>>> Handle(GetAllUserNotificationsQuery request, CancellationToken cancellationToken)
+    {
+        var operationResult = new OperationResult<IEnumerable<UserNotification>>();
+
+        var userNotifications =
+            await _userNotificationRepository.GetAllAsync(cancellationToken);
+
+        if(!userNotifications.Any())
+            return operationResult.ReturnWithErrorMessage(new NotFoundException(
+                "No notifications were found!"));
+
+        var allNotificationForCurrentUser = userNotifications
```

Comment:
This is not optimal. Why? 

Because we are getting all userNotifications and then filtering only the userNotifications that we need foir the current user.

It came the moment to add one more method with predicate. To filter in database only the items that we need.

Where..... clause with expression

Added in the repository

Reply:
Done.